File: game/forest/hurt_flying_talking.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.hurt_flying_falling import HurtFlyingFalling
from state import State


class HurtFlyingTalking(State):
    def process_events(self, events):
        if self.get_frame_index() >= len(ForestResources.sync_catapult_talktop):
            return HurtFlyingFalling

        return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.speak_catapult_talktop, self.context.country)

    def render(self, screen):
        screen.blit(Animation.get_sync_frame(ForestResources.catapult_airtalk, ForestResources.sync_catapult_talktop, self.get_frame_index()), (0,0))

File: game-web/src/forest/hurtFlyingTalking.ts
/** Port of game/forest/hurt_flying_talking.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { HurtFlyingFalling } from './hurtFlyingFalling';
import type { PhoneEvents } from '../phoneEvents';

export class HurtFlyingTalking extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndex() >= ForestResources.syncCatapultTalktop.length ? HurtFlyingFalling : null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.speakCatapultTalktop);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getSyncFrame(ForestResources.catapultAirtalk, ForestResources.syncCatapultTalktop, this.getFrameIndex()), 0, 0);
  }
}
